pyacq/core/sharedarray.py

```diff
@@ -55,13 +43,62 @@ def __init__(self, shape=(1,), dtype = 'float64', shm_id = None):
                 self.mmap = mmap.mmap(self._tmpFile.fileno(), self.nbytes, mmap.MAP_SHARED, mmap.PROT_READ)
                 
     def close(self):
+        """Close this buffer.
+        """
         self.mmap.close()
         if not sys.platform.startswith('win') and hasattr(self, '_tmpFile'):
             self._tmpFile.close()
     
     def to_dict(self):
-        return {'shape': self.shape, 'dtype': self.dtype, 'shm_id': self.shm_id}
+        """Return a dict that can be serialized and sent to other processes to
+        access this buffer.
+        """
+        return {'nbytes': self.nbytes, 'shm_id': self.shm_id}
+    
+    def to_numpy(self, offset, dtype, shape, strides):
+        """Return a numpy array pointing to part (or all) of this buffer.
+        """
+        return np.ndarray(buffer=self.mmap, shape=shape,
+                          strides=strides, offset=offset, dtype=dtype)        
+        
+
+
```

Comment:
Not at all! I think you explained it clearly many times, but I kept seeing this:

    [1111, 2222] + [1111, 2222] = [1111, 2222, 1111, 2222]

(which doesn't make any sense) instead of this:
 
    [1111, 2222] + [1111, 2222] = [11111111, 22222222]
 
I might revert back to the point where I had a fully functional ND ring buffer. And I finally almost agree that we need `swapaxes`  :D